pkgs/development/tools/java/java-language-server/default.nix

```diff
@@ -0,0 +1,100 @@
+{ lib, stdenv, fetchFromGitHub
+, jdk, maven, runtimeShell
+}:
+
+let
+  platform =
+    if stdenv.isLinux then "linux"
+    else if stdenv.isDarwin then "mac"
+    else if stdenv.isWindows then "windows"
+    else throw "unsupported platform";
+in
+stdenv.mkDerivation rec {
+  pname = "java-language-server";
+  version = "0.2.38";
+
+  src = fetchFromGitHub {
+    owner = "georgewfraser";
+    repo = pname;
+    # commit hash is used as owner sometimes forgets to set tags. See https://github.com/georgewfraser/java-language-server/issues/104
+    rev = "1dfdc54d1f1e57646a0ec9c0b3f4a4f094bd9f17";
+    sha256 = "sha256-zkbl/SLg09XK2ZhJNzWEtvFCQBRQ62273M/2+4HV1Lk=";
+  };
+
+  fetchedMavenDeps = stdenv.mkDerivation {
+    name = "java-language-server-${version}-maven-deps";
+    inherit src;
+    buildInputs = [ maven ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      mvn package -Dmaven.repo.local=$out/.m2 -DskipTests
```

Comment:
This is slightly different from the example at https://nixos.org/manual/nixpkgs/unstable/#double-invocation - any reason to use `$out/.m2` rather than just `$out`?

Reply:
Looking at it too, none that I could think of. (Perhaps its an artifact of when I was having issues trying to get it to build) Ill change it to just `$out`. 

Also, the reason to skip the tests, the [build instructions](https://github.com/georgewfraser/java-language-server#vim-with-vim-lsc)  say to skip them  and indeed, I was getting an error when I let them run. (Ill attempt to build it to show you the error)

Reply:
I ran the tests and I believe all of the errors were caused by a module not being exported
```
java.lang.IllegalAccessError: superclass access check failed: class org.javacs.ReusableCompiler$ReusableContext (in unnamed module @0xb6a0f7) cannot access class com.sun.tools.javac.
util.Context (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.util to unnamed module @0xb6a0f7
```
I believe this problem could be mitigated by adding 
```
<arg>--add-exports</arg>
<arg>jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
```
to each of the dependency compilerArgs in `pom.xml` but ... ill wait to see what you say